lib/rules/function-calc-no-unspaced-operator/index.mjs

```diff
@@ -222,21 +135,14 @@ const rule = (primary, _secondaryOptions, context) => {
 					const token = operation.firstOperand.value;
 
 					if (isTokenDimension(token)) {
-						const { unit } = token[4];
-						const operatorChar = unit.slice(-1);
+						const [, , , endPos, { unit }] = token;
+						const operatorChar = unit.at(-1);
 
 						if (operatorChar && OPERATOR_REGEX.test(operatorChar)) {
-							fixOperationWithoutOperator(operation, token, operatorChar);
+							operation.completeMissingOperator(operatorChar, endPos, 'append');
 
 							if (context.fix) {
```

Comment:
[note] Note that `context.fix` is still necessary because it's impossible to call `report()` at this point. After completing a missing operator, we need to insert whitespaces by autofix.

Reply:
I was looking at this rule too and noticed this aswel.

My thinking at the time was that one function can extract and restore the operator.
And another function can be dedicated to fixing whitespace.

But maybe better to refactor this so that each mutation fully fixes each error?

Reply:
Sounds good. Can you push a commit to this PR or create a new PR?

Reply:
I don't have much time today, but hopefully tomorrow I can look into this :)

(separate PR is maybe is easier? I'll see how large the change is)

Reply:
Thanks. I'm fine with either. As you said, this PR is already quite big. 😅

Reply:
> Note that `context.fix` is still necessary…

@ybiquitous When I looked at this refactor, the only solution that didn't refactor the rule from the ground up was to have an array that would collect functions which all would be called in a `forEach` inside the fixer.

Reply:
@Mouvedia, your idea sounds possible, but I'm still unsure if it works. So, let's wait for @romainmenke. He might bring a better solution.

Reply:
This was resolved in https://github.com/stylelint/stylelint/pull/7835